### ISSUE TITLE
The Romanian plural formula in poedit is wrong

### ISSUE THREAD

**Issue by dasj19** [vslavik/poedit#465]
Hello, one of our new translators from Gnome noticed that poedit warns the user about our project (The Gnome Project: https://l10n.gnome.org/teams/ro/ ) using a wrong plural formula.
But this is not the case.
The plural formula for Romanian embedded in poedit is wrong.

The plural formula currently used in poedit is:

nplurals=3; plural=(n==1?0:(((n%100>19)||((n%100==0)&&(n!=0)))?2:1));

The correct formula is: 

nplurals=3; plural=(n==1 ? 0 : (n==0 || (n%100 > 0 && n%100 < 20)) ? 1 : 2);

Some sources: 

[1] http://docs.translatehouse.org/projects/localization-guide/en/latest/l10n/pluralforms.html?id=l10n/pluralforms

[2] https://developer.mozilla.org/en-US/docs/Mozilla/Localization/Localization_and_Plurals (Rule no.5)

[3] https://www.gnu.org/software/gettext/manual/html_node/Plural-forms.html (Search for Romanian)

**Comment by secarica** [comment]
This is also related to this past message
https://groups.google.com/forum/#!msg/poedit/Kpq3ihRxXrM/-dLe1p1RSpwJ

**Comment by vslavik** [comment]
Would you know for reference what `cldr-plurals` produces?

**Comment by vslavik** [comment]
> Would you know for reference what cldr-plurals produces?

Thanks for helping me out here... 😞 To answer my own question, GNU gettext official tool for this purpose generates different, but equivalent
```
nplurals=3; plural=(n==1 ? 0 : (n==0 || (n%100 > 0 && n%100 < 20)) ? 1 : 2);
```

**Comment by secarica** [comment]
This formula was already mentioned on top of this very thread (> The correct formula is: etc.).

As for CLDR, when it comes to GNU gettext, I'd rather trust more the native GNU documentation (whose link has also been indicated on top of this very thread).

**Comment by vslavik** [comment]
> This formula was already mentioned on top of this very thread

I made a — now corrected — copy & paste mistake. The point is, there are other forms in use too.

> As for CLDR, when it comes to GNU gettext, I'd rather trust more the native GNU documentation 

I'm afraid you are dismissing me from the position of ignorance. `cldr-plurals` is a *GNU gettext* tool. It is included in default install and used my `msginit` to initialize `Plural-Forms` correctly for a language — it is what GNU gettext regards as the source of correct information. Unlike the few *examples* in years-untouched docs, it is comprehensive and current.

**Comment by vslavik** [comment]
The issue here is many-fold:

1. Wrong formula for a smallish (well, twice as large as *mine*…) language. This is, frankly, the least worrisome bit — it's a small, slightly annoying, bug affecting a few users.

2. Poedit's equality check is touchy and complains about both some formatting differences and equivalent, but differently expressed, plural form expressions.

3. It is not feasible to manually curate the list, for all languages (case in point: the limited coverage of the GNU manual's list). It needs to be automated. Which Poedit does. The problem is, it uses a [terrible source](https://github.com/vslavik/poedit/blob/217488705f019b88536ccf4877a25f0cb0de8ce8/scripts/extract-plural-forms.py) — unmaintained, dead open source version of Transifex.

Consequently, the fix is more involved as well:

1. Hot-patch Romanian manually for 2.0.8. That takes care of this issue, but not the underlying one.
2. Change to a better source of data. *Everybody* uses CLDR these days and because GNU gettext does too (see above), so will Poedit.
3. The equivalence check must be more robust and should treat the above two correct formulas ("traditional" one from GNU gettext manual as used by GNOME, and CLDR-generated one) as same and not warn about using one instead of the other.

**Comment by vslavik** [comment]
…And with 61d4394388737fb6228c668d622a04bec0426d53 this is fixed properly for 2.1.

**Comment by dasj19** [comment]
@vslavik Thanks for solving this issue, looking forward to work with poedit
